Backend.php

```diff
@@ -0,0 +1,26 @@
+<?
+
+namespace iFixit\TokenBucket;
+
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'StoredBucket.php';
+
+interface Backend {
+   /**
+    * Value for when there was no key found in storage.
+    */
+   const MISS = null;
+
+   /**
+    * Retrieve the value of the given string key.
+    * Needs to return an StoredBucket.
+    *
+    * should return a StoredBucket
+    */
+   public function get($key);
+   /**
+    * Set the value of the given string key to whatever type value is.
+    *
+    * Expiration time is given as an integer number of seconds.
```

Comment:
Yeah, so if someone wants to, they can have their Backend implentor use a unix timestamp for invalidation, or a time from now to invalidate.

Yeah, you're right, the stored bucket doesn't need to keep track of when it should be expired.

This could be done either way, and then the backend can decide what to do with it, but since we use Memcache, and this is what memcache normally takes, that's how I made it.